earthfile2llb/converter.go

```diff
@@ -724,6 +727,9 @@ func (c *Converter) internalRun(ctx context.Context, args []string, secretKeyVal
 	runEarthlyMount := llb.AddMount("/run/earthly", llb.Scratch(),
 		llb.HostBind(), llb.SourcePath("/run/earthly"))
 	finalOpts = append(finalOpts, debuggerSecretMount, debuggerMount, runEarthlyMount)
+	if c.withSSH {
```

Comment:
Wondering if we should have this as a setting that needs to be enabled in the Earthfile, similarly to how buildkit requires [`--mount=type=ssh`](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/experimental.md#run---mounttypessh).

Wondering how this might translate to the cloud runner eg `--mount=type=ssh,id=+secrets/<org>/path/ssh-key`.. or something like that.

Reply:
That's a good question.

So if a user has the following in their Earthfile
`RUN --ssh <cmd that needs ssh>` we could proxy through to the specified (or default) ssh socket agent; but obviously that won't work for the cloud runner.

So then if we allow a value to be specified like:
`RUN --ssh=+secrets/<org>/path/ssh-key <cmd that needs ssh>` then it would grab the ssh key instead.

and then if there's a basic RUN command without an ssh flag, then the ssh socket won't be made available to the command?

Reply:
Yes, something along those lines.

(BTW, adding a new flag to RUN is easy: https://github.com/earthly/earthly/blob/master/earthfile2llb/listener.go#L266 )

> and then if there's a basic RUN command without an ssh flag, then the ssh socket won't be made available to the command?

I would think so.

Reply:
I added in a `--ssh` flag, and down the road that'll give us the flexibility to hook it into the secrets.